### PR TITLE
fix: update password reset to use lowercase emails

### LINKED ISSUE #1516: Password reset email needs to be case insensitive
Password recovery is not case insensitive, although the email check is. The entire process needs to be lowercased at all steps.

### PR DESCRIPTION
**Pull request by malexanderboyd**
Fixes #1516 

- refactored all references to `req.body.email` to `recoveryEmail` and cast to lowercase.

issue-1516

### CODE REVIEW

File: routes/users_routes.js [hunk 1]
```diff
@@ -235,7 +236,7 @@ router.post(
           } else {
             User.findOne(
               {
-                email: req.body.email,
+                email: recoveryEmail,
```
Comment (malexanderboyd) [inline routes/users_routes.js:239]: This is the only one I wasn't 100% with - should we use the email exactly how the user types it or downcast it? Not sure if `findOne` is case-insensitive under the covers.

Comment (dekkerglen) [reply routes/users_routes.js:239]: We should use findOne without the regex, as it is O(1) vs O(n). As long as we consistently normalize on write and on read, it won't be an issue

Comment (malexanderboyd) [reply routes/users_routes.js:239]: Makes sense! 

Not sure how long the writes have been normalized to lowercase (new contributor), but for any user that somehow registered without being normalized to lower case would be unable to recover their password following this?

Comment (dekkerglen) [reply routes/users_routes.js:239]: No, the login recovery documents are only valid for 15 min. We already normalize emails for the User table

**Pull request merged**